### DISCUSSION TITLE
[OTBR][MLE]OTBR Connection Probability Issue with Identical Parameters

### DISCUSSION THREAD

**Discussion by aaa147147** [openthread/openthread#11095] [General]
**OTBR-1 Parameters**
```
Active Timestamp: 1
Channel: 17
Wake-up Channel: 12
Channel Mask: 0x07fff800
Ext PAN ID: 223bf7b2c0121211
Mesh Local Prefix: fd52:b7bd:20e5:53d::/64
Network Key: 1030ba7ebf053d52c12101c1b286b54b
Network Name: 1234
PAN ID: 0x15f2
PSKc: f980bbb18cde32da1b1a1309e0af0cc0
Security Policy: 672 onrc 0
```
**OTBR-2 Parameters**
```
Active Timestamp: 1
Channel: 17
Wake-up Channel: 23
Channel Mask: 0x07fff800
Ext PAN ID: 223bf7b2c0121211
Mesh Local Prefix: fd52:b7bd:20e5:53d::/64
Network Key: 1030ba7ebf053d52c12101c1b286b54b
Network Name: 1234
PAN ID: 0x15f2
PSKc: f980bbb18cde32da1b1a1309e0af0cc0
Security Policy: 672 onrc 0
```
**During our testing, we found that the two devices mentioned above had a probability of being unable to connect and cleared. Some MLE logs are as follows:**

**OTBR-1**
![image](https://github.com/user-attachments/assets/8a4b5455-22b8-49fc-9a6c-f2fd90254bff)
**OTBR-2**
![image](https://github.com/user-attachments/assets/2afde8af-3a9e-442f-a544-5bfee4e7a0d3)

**Full Log**
**OTBR-1**
[1.log](https://github.com/user-attachments/files/18265341/1.log)
**OTBR-2**
[2.log](https://github.com/user-attachments/files/18265342/2.log)

**Comment by aaa147147** [comment]
**Can we get any useful information from this to determine where the problem lies?**

**Comment by jwhui** [comment]
It's not immediately clear why OTBR-1 is failing to process the Child ID Request with InvalidState error.

Some possibilities:

https://github.com/openthread/openthread/blob/2ae1ce52931bbb887e43a55937a3210d14012b3d/src/core/thread/mle_router.cpp#L2018

This is not likely since there is a similar check in `HandleParentRequest()`.

https://github.com/openthread/openthread/blob/2ae1ce52931bbb887e43a55937a3210d14012b3d/src/core/thread/mle_router.cpp#L2020

This is not likely since there is a similar check in `HandleParentRequest()`.

https://github.com/openthread/openthread/blob/2ae1ce52931bbb887e43a55937a3210d14012b3d/src/core/thread/mle.cpp#L4920

This occurs when the device is still trying to retrieve the Thread Network Data. But it's not clear to me why the device would be in this state indefinitely.

Are you able to provide logs that include the period when OTBR-1 attaches to the Thread network? That may provide more visibility into how OTBR-1 arrived in this state.

**Reply by aaa147147** [reply]
@jwhui Thank you for your reply. The probability of this issue occurring is very small. The next time it occurs, we will rewrite the provided log
